pkg/cache/kubeConfig.go

```diff
@@ -0,0 +1,82 @@
+package cache
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/jellydator/ttlcache/v3"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+var kubeConfigCache = ttlcache.New[string, string](
+	ttlcache.WithTTL[string, string](5*time.Minute),
+	ttlcache.WithDisableTouchOnHit[string, string](),
+)
+
+// GetKubeConfigFromCache returns the kubeconfig from the cache if it is not expired.
+// If it is expired, it will get the kubeconfig from the secret and set it in the cache.
+func GetKubeConfigFromCache(logger *zap.SugaredLogger, clientSet *kubernetes.Clientset, runtimeID string) string {
+	kubeConfigCache.DeleteExpired()
```

Comment:
Out of curiosity: Is it necessary? Doesn't the cache evict items automatically/internally?

Reply:
:+1: